akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala

```diff
@@ -338,6 +338,9 @@ trait ActorRefFactory {
   /**
    * Stop the actor pointed to by the given [[akka.actor.ActorRef]]; this is
    * an asynchronous operation, i.e. involves a message send.
+   * If this method is applied to the `self` reference from inside an Actor
+   * then that Actor is guaranteed to not process any further messages before
```

Comment:
then why don't you just say that it is guarenteed to not process any further message *after this call*